lib/paths-cache.js

```diff
@@ -223,13 +224,30 @@ export default class PathsCache extends EventEmitter {
     this._cancelled = false
     this.emit('rebuild-cache')
 
-    return execPromise('find --version')
-      .then(
+    // check if find is available and store its path
+    if (this.hasFind == null) {
+      which('find')
+      .then( (resolvedPath) => {
+        this.findPath = resolvedPath
+        this.hasFind = true
+      })
+      .catch( () => {
+        this.hasFind = false
+      })
+    }
+
+    try {
+      if (this.hasFind) {
         // `find` is available
-        () => this._buildInitialCacheWithFind(),
+        return this._buildInitialCacheWithFind()
+      } else {
         // `find` is not available
-        () => this._buildInitialCache()
-      )
+        return this._buildInitialCache()
+      }
+    } catch {
+      // fallback
+      return this._buildInitialCache()
```

Comment:
It seems strange to run `this._buildInitialCache()` twice if it fails the first time.

This catch should only be around `this._buildInitialCacheWithFind()`

Reply:
Oh, yes thanks

Reply:
https://github.com/atom-community/autocomplete-paths/pull/245/commits/80449344d17d815de7a380c8b71a16ff4fa33d9b